stream.go

```diff
@@ -312,6 +316,22 @@ func (s *Stream) Close() error {
 	s.stateLock.Unlock()
 	return nil
 SEND_CLOSE:
+	// This shouldn't happen (the more realistic scenario to cancel the
+	// timer is via processFlags) but just in case this ever happens, we
+	// cancel the timer to prevent dangling timers.
+	if s.closeTimer != nil {
+		s.closeTimer.Stop()
+		s.closeTimer = nil
+	}
+
+	// If we have a StreamCloseTimeout set we start the timeout timer.
+	// We do this only if we're not already closing the stream since that
+	// means this was a graceful close.
```

Comment:
I think I would appreciate as a future reader a example scenario this avoids.

"This prevents a leaked buffer if one side of the stream closes, but the other does not. After the specified timeout, the stream is force closed."

Reply:
Good, expanded on the comment.  Done.